engine/src/cubos/engine/gizmos/gizmos.cpp

```diff
@@ -16,13 +16,29 @@ using cubos::engine::GizmosRenderer;
 
 using namespace cubos::core::gl;
 
-Gizmos::Gizmo::Gizmo(const std::string& id, glm::vec3 color, float lifespan)
-    : mId(id)
+Gizmos::Gizmo::Gizmo(uint32_t id, glm::vec3 color, float lifespan)
+    : id(id)
     , mColor(color)
     , mLifespan(lifespan)
 {
 }
 
+void Gizmos::registerInputGizmo(uint32_t id, bool pressed)
+{
+    mIdInteraction = id;
+    mPressed = pressed;
+}
+
+bool Gizmos::pressed(const std::string& id) const
+{
+    return mPressed && mHasher(id) == mIdInteraction && mIdInteraction != 0;
+}
+
+bool Gizmos::hover(const std::string& id) const
+{
+    return !mPressed && mHasher(id) == mIdInteraction && mIdInteraction != 0;
```

Comment:
I don't really like the combination of using a hash as ID and also having magic values such as 0. But its okay to leave this as is. When we generalize screen picking to all kinds of entities, we'll have to change the logic on how we identify stuff anyway.

Reply:
I don't like it either, but there's no reasonable way to store strings on a texture, and using integer identifiers could easily lead to even more collisions

Reply:
We can easily use integer identifies by storing a map of strings to integers on the CPU side, but yeah, that isn't necessary rn.